.circleci/config.yml

```diff
@@ -38,15 +38,15 @@ jobs:
 
   lint:
     docker:
-      - image: circleci/golang:1.14
+      - image: cimg/go:1.18
     steps:
       - checkout
 
       - restore_cache:
           keys:
             - v1-pkg-cache
 
-      - run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/fc0a898a6ae297c0ef59e2f1d824713d6f1cd222/install.sh | sh -s -- -d -b $(go env GOPATH)/bin v1.27.0
+      - run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/a31485fe906612b2884239fdf7c55aec76ea7596/install.sh | sh -s -- -d -b $(go env GOPATH)/bin v1.50.1
```

Comment:
The old version no longer worked and was consistently failing on CI so this needed to be upgraded as well